### DISCUSSION TITLE
Access to storage the usual way

### DISCUSSION THREAD

**Discussion by vlebert** [opengisch/QField#2811] [Q&A]
Hi

Last update (2.0.15) totally breaks our workflow because of the new way to access local projects ("importing" instead of direct opening).
As we began far before Qfield Cloud, we use our own mechanisms (based previously on [syncthing](https://syncthing.net/) and now on GIT) to handle projects and bidirectional syncing.

My company (Tactis) is user of Qfield since 2017, and we have around 20 tablets using daily Qfield.

I understand it is related to android security policy... Is there really no workaround ? I noticed that "recent" shortcuts are still working without importing the projects.

At the moment we are forced to use older version because our workflow is not compatible with this change...

I'd like to investigate more Qfield Cloud, but it is still "beta", and we have really heavy use with 10's of Go of pictures etc... We are to much "in a rush". I'll think about if when I will be sure it can be a solution for our day to day production.

Regards
Valérian LEBERT

**Comment by nirvn** [comment]
@vlebert , have you read the storage documentation (https://docs.qfield.org/get-started/storage/) yet? If so, can you elaborate on the specific workflow(s) your company is having a hard time adapting? That'd be quite helpful in building an argument with Google to put QField in an arbitrary whitelist to regain full storage access (as some apps have already benefited from). Note that it is very likely that Google will sooner or later get rid of this full storage access whitelist for apps, which would bring us to the same point :)

The "recent" shortcuts are merely shortcuts to projects you would have imported to begin with. If you are on Android >= 10, any lingering project shortcut that's not referring to an imported project/dataset wouldn't work.

**Comment by nirvn** [comment]
(@vlebert , I've moved this into a slightly more appropriate discussion format)

**Comment by vlebert** [comment]
Hi,

regarding our workflow, we use 2 different method : 

**Method 1 :**
- We use Seafile (with Seadroid version <=2.25) for project "repository". Tablets can download and update projects from here (direction master --> user tablet)
- We use [Syncthing ](https://syncthing.net/)for uploading and consolidating field data (point layer + pictures) (direction tablets --> master server)

Note : we had to block Seadroid to version 2.25 for the same reason (neweer version sync data in a private android folder somewhere lost in the filesystem)

From there we have routines on the server hosting Syncthing and Seafile to aggregate the data and export it in several ways (Postgis, etc.)

**Method 2 :**
- We use termux, termux-api, termux-gui, GIT and customs scripts to act as a simple GUI allowing tablets to sync projects from master
- Gitlab hosting (10Go per repo)
- Master branch for project updates and 1 branch per tablet for field data and pictures upload (with resize within termux)


For french readers, I drafted an article aimed to be published on https://static.geotribu.fr/ (work inprogress)
https://digi-studio.notion.site/Relev-terrain-avec-Qfield-f24df92fcbb94847aafcd687ab9af859
There are more details inside

About the "recent" shortcut, I am surprised because I updated to new Qfield version without importing any project dans have a "recent" shortcut to an external project and seems fully functional (so no problem accessing full storage ?)

Valérian

**Reply by nirvn** [reply]
@vlebert , on android <= 9, you would retain same storage access as before, maybe your device is older and runs on that version of android?

**Reply by vlebert** [reply]
Galaxy tab S4 / Android 10

**Reply by nirvn** [reply]
@vlebert , feel free to attach a screencast if you want this further dissected :)

**Reply by vlebert** [reply]
![qfield](https://user-images.githubusercontent.com/49779177/165067345-e0d6b785-baa5-45bd-8e3c-e5b740349cff.gif)

Does this replies to  you need : no trace of imported project and still access to 1 recent project

**Reply by nirvn** [reply]
Where is this project located on your storage?

**Reply by vlebert** [reply]
Internal Storage/Seafile/

**Reply by nirvn** [reply]
@vlebert , ah, can you check something for me: are you able to _write_ to the datasets there, or merely read from it?

**Reply by vlebert** [reply]
Read-only indeed. Error when trying to write a feature

**Reply by nirvn** [reply]
@vlebert , ok thanks, we should hide those upon upgrades to avoid confusion.

With the new export folder functionality added in 2.0.15 which allows you to copy the content of your project into a folder on your storage that's accessible by seafile and synchthing, is the problem there that it adds and extra export step that's too much of a burden to your field workers?

**Reply by vlebert** [reply]
Thanks for the advice, however I can't imagine this solution to work for field workers.
Syncing must be quite transparent as these workers are not always very familiar with android etc.

**Reply by nirvn** [reply]
@vlebert , fair enough. So you guys are setting up the services devices and provide them to field workers, some of whom might have limited knowledge on IT matters and can only be taught to enter data and not much else. Is that a fair representation?

**Reply by vlebert** [reply]
It is.
We provide some support with Teamviewer but it is quite painful.

**Reply by nirvn** [reply]
Thanks @vlebert , helpful information. 

One last question: would manual instalation of APKs will full storage access an acceptable solution for your team who set up the device prior to deployment? The downside would mean no auto update through the play store but in your case you might not want this to begin with to create a static environment while your field workers hold onto their devices.

**Reply by vlebert** [reply]
That's what we do : manual install from release github page
Then set playstore autoupdate "no" for Qfield.

However, sometimes if you go to playstore and manually click to "all updates" then playstore will ingnore your preferences for specific app and update Qfield anyway.

Even if you install from APK, playstore will somehow link it to the official Qfield. Can this behaviour be changed?

**Reply by nirvn** [reply]
There are ways around avoiding this type of autoupdate. 

Thanks for taking the time to provide detailed information on your workflow. We'll look into what can be done.

In the meantime, I can't stress enough of good a solution QFieldCloud might be for you, please take some time to explore it. It has to be the smoothest and most secure remote deployment there is for QField. There are nice users management (multi user project access and data push, user roles, etc.) features too that could make your workflows even better than before.

**Reply by vlebert** [reply]
Yes I will give a try because our current workflow is reaching its limits. But I am really concern about the "coming soon". It's been some years it's coming soon 🙃
![image](https://user-images.githubusercontent.com/49779177/165108259-9afa767c-8791-43fa-a818-a3776493712e.png)

What is the current storage limit ? We produce tons of JPEG... And our current workflow allow us to resize JPEGs to lower res wich in not possible in Qfield and even in natives camera apps (samsung at least)...

**Reply by vlebert** [reply]
And what ways were you thinking to avoid autoupdate?

**Reply by vlebert** [reply]
Would you have a moment to have a quick call to check if Qfield Cloud could respond to all our needs technically ? 
And then economically if this goes further

I already have a few questions after a quick try :
- Spatialite support ? 
- Spatial views support (gpkg or spatialite. I am not sure if it is allowed by gpkg)
- Access "machine-to-machine" to the synced files to automate workflows

Of course we could also change our workflow and stop using spatialite :)

Anyway I would be glad to discuss this and see if time has come so switch to QCloud

**Reply by nirvn** [reply]
@vlebert , we'd be more than happy to chat about QFieldCloud and its capabilities. Please contact Marco Bernasocchi (marco@opengis.ch), our passionate CEO, he'll be able to talk you through this.

On the technical questions you've raised above:
- Regarding JPEG resizing, QField 2.1 actually supports that, it's one of the nice new features of that release. You set the maximum image width/height in a new project-wide settings when preparing your project in QGIS + qfieldsync.
- Spatialite datasets as well as spatial views are supported by QFieldCloud.
- Spatial views are not only supported by geopackages, but would definitively work better in this environment.

**Comment by ghost** [comment]
> Thanks @vlebert , helpful information.
> 
> One last question: would manual instalation of APKs will full storage access an acceptable solution for your team who set up the device prior to deployment? The downside would mean no auto update through the play store but in your case you might not want this to begin with to create a static environment while your field workers hold onto their devices.

Just wanted to say that the availability of manually installable APKs with full storage access would be really useful for us too. Are you thinking they'd be a commercial product, or part of the FOSS offering?

**Comment by vlebert** [comment]
The APKs are here
https://github.com/opengisch/QField/releases

The only annoying part is that the APK still tries to update to the latest one via playstore. You have to block autoupdate and be vigilent in manual updates

**Reply by ghost** [reply]
Thanks. That's actually the solution I'm currently using (an old APK with GooglePlay updates turned off) I was more here wondering if the Qfield team might see their way clear to occasionally releasing _updated_ APKs with the storage access still available as an option for those who don't care about Google Play's rather overbearing restrictions.

**Comment by vlebert** [comment]
Any news regarding QField beein on a white list to access local storage ?
Newer version are unusable for us and though QField Cloud seems an excellent product, it is also not ready for production for us while because we don't manage to use the API and the documentation is very light.
